### PR TITLE
Release/DIMSv3.1.0

### PR DESCRIPTION
**Pull request by ALuesink**
Changes for the DIMS HMDB update:

- #94 

During validation and test runs some errors occurred, so the last 5 commits are new commits and have not been reviewed in the previous PR.

Commits [32b532a](https://github.com/UMCUGenetics/DIMS/pull/95/commits/32b532afa33af6fb50aace73428c48d9359b97af) & [4b621c1](https://github.com/UMCUGenetics/DIMS/pull/95/commits/4b621c1b8e5f32b0548d4914eb24004edfc33df4) contain new DIMS repo code, the other new commits are CustomModules commits.

### CODE REVIEW

**Review by melferink — DISMISSED** [review]
All commit from PR have been appoved. New commits are ok!

**Review by melferink — DISMISSED** [review]
New CustomModules version is correct

File: README.md [hunk 1]
```diff
@@ -4,30 +4,22 @@ Pipeline that processes raw Direct Infusion Mass Spectrometry data.
 ### Folder Structure
 ```
 .
-|───GUI/ (GUI scripts)
+|───CustomModules/ (GitHub repo with pipeline scripts)
+|───assests/ (extra Nextflow files)
```
Comment (melferink) [inline README.md:8]: typo! assets

File: README.md [hunk 2]
```diff
@@ -4,30 +4,22 @@ Pipeline that processes raw Direct Infusion Mass Spectrometry data.
...
+docker build -t ghcr.io/umcugenetics/[NAME]:[tag] .
+docker push ghcr.io/umcugenetics/[NAME]:[tag]
 ```
 
 on HPC: 
 ```
 srun -c 2 -t 0:30:00 -A dbg_mz --mem=100G --gres=tmpspace:100G --pty /usr/bin/bash 
 cd /hpc/dbg_mz/tools/singularity_cache/ 
-singularity build /hpc/dbg_mz/tools/singularity_cache/dims-[tag].img docker://umcugenbioinf/dims:[tag]
+singularity build /hpc/dbg_mz/tools/MAP/NAME.img docker://ghcr.io/umcugenetics/[NAME]:[tag]
```
Comment (melferink) [inline README.md:22]: /hpc/dbg_mz/tools/MAP/ does not exist on HPC. Is MAP a variable that needs to be changed by the user? Maybe include [MAP] /< MAP> and explain a bit about these variables

File: README.md [hunk 3]
```diff
@@ -42,56 +34,50 @@ Libraries: xcms, stringr, dplyr, Rcpp, openxlsx, reshape2, loder, ggplot2, gridE
...
 
 ## Usage
-The pipeline is meant to be started with the GUI, which is an R shiny program to transfer data to the HPC and start the pipeline. To open the GUI, open GUI.Rproj in Rstudio, which should open run.R and config.R. Then click "Run App" from the run.R file. 
+The pipeline can be started with a GUI, which is an R shiny program to transfer data to the HPC and start the pipeline. The GUI access can only be used when someone has access. To get access contact the bioinformaticians. 
 
 Manually starting the pipeline is also possible.
 ```
 CMD:
-  sh run.sh -i <input path> -o <output path> [-r] [-v] [-h]
+/hpc/dbg_mz/production/DIMS/run_nextflow_dims.sh -i <input path> -o <output path> -e <email> -s <samplesheet> -n <nr_replicates> -r <resolution> -p <ppm> -z <zscore> -m <matrix> -t <standard_run> [-v] [-h]
```
Comment (melferink) [inline README.md:58]: "sh <>" not needed anymore for run_nextflow_dims.sh/run.sh?

File: README.md [hunk 4]
```diff
@@ -42,56 +34,50 @@ Libraries: xcms, stringr, dplyr, Rcpp, openxlsx, reshape2, loder, ggplot2, gridE
...
-  -o - full path output folder, e.g. /hpc/dbg-mz/processed/run1
+    -i - full path input folder, eg /hpc/dbg_mz/raw_data/run1 (required)
+    -o - full path output folder, eg. /hpc/dbg_mz/processed/run1 (required)
+    -e - emailadress, eg. user@umcutrecht.nl (required)
+    -s - samplesheet, eg. sampleNames.txt (required)
+    -n - number of replicates, eg. 2 (required)
+    -r - resolution, eg. 140000 (required)
+    -p - ppm, eg. 5 (required)
+    -z - zscore, 1 for Z-score and 0 for no Z-score (required)
+    -m - matrix, eg. Plasma (required)
```
Comment (melferink) [inline README.md:69]: Maybe too much for now but:  is the a free-field, or are there restricted options here? If so, you could mention them here.

**Review by melferink — CHANGES_REQUESTED** [review]
Few small comments

File: README.md (hunk 1 above)
Comment (ALuesink) [reply README.md:8]: Thanks, is changed.

File: README.md (hunk 2 above)
Comment (ALuesink) [reply README.md:22]: Thanks, this was a copy error and has been changed.

File: README.md (hunk 3 above)
Comment (ALuesink) [reply README.md:58]: The run_nextflow_dims.sh file has a shebang at the top of the script and by calling the complete location of the script you don't need "sh" to start the script. If you don't use the complete location of the script, you do need to use "sh" before the bash file.

File: README.md (hunk 4 above)
Comment (ALuesink) [reply README.md:69]: Good comment. At the moment these are free-fields, but some could be changed to restricted options. I'll add it for a future refactor.

File: README.md [hunk 5]
```diff
@@ -42,56 +34,50 @@ Libraries: xcms, stringr, dplyr, Rcpp, openxlsx, reshape2, loder, ggplot2, gridE
...
+    -t - standard run, yes or no (required)
 
 OPTIONAL ARGS:
-  -r - restart the pipeline, removing any existing output for the entered run (default off)
   -v - verbose printing (default off)
   -h - show help
 
 EXAMPLE:
-  sh run.sh -i /hpc/dbg_mz/raw_data/run1 -o /hpc/dbg_mz/processed/run1
+  /hpc/dbg_mz/production/DIMS/run_nextflow_dims.sh -i /hpc/dbg_mz/raw_data/run1 -o /hpc/dbg_mz/processed/run1$ -e user@umcutrecht.nl -s sampleNames.txt -n 2 -r 140000 -p 5 -z 1 -m Plasma -t yes
```
[the remark this answers is not in the document]
Comment (ALuesink) [reply README.md:77]: See previous reply.

**Review by melferink — APPROVED** [review]
Ok!

**Pull request merged**